ignite/services/chain/init.go

```diff
@@ -164,11 +177,99 @@ func (c *Chain) InitAccounts(ctx context.Context, cfg *chainconfig.Config) error
 	c.ev.SendView(accounts, events.ProgressFinish())
 
 	// 0 length validator set when using network config
-	if len(cfg.Validators) != 0 {
-		_, err = c.IssueGentx(ctx, createValidatorFromConfig(cfg))
+	if len(cfg.Validators) == 0 {
+		return nil
 	}
+	if cfg.IsConsumerChain() {
+		// Consumer chain writes validators in the consumer module genesis
+		if err := c.writeConsumerGenesis(); err != nil {
+			return err
+		}
+	} else {
+		// Sovereign chain writes validators in gentxs.
+		_, err := c.IssueGentx(ctx, createValidatorFromConfig(cfg))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
 
-	return err
+// writeConsumerGenesis updates the consumer module genesis in the genesis.json
+// file of c.
+func (c *Chain) writeConsumerGenesis() error {
+	var (
+		providerClientState = &ibctmtypes.ClientState{
+			ChainId:         "provider",
+			TrustLevel:      ibctmtypes.DefaultTrustLevel,
+			TrustingPeriod:  time.Hour * 64,
+			UnbondingPeriod: time.Hour * 128,
+			MaxClockDrift:   time.Minute * 5,
+		}
+		providerConsState = &ibctmtypes.ConsensusState{
+			Timestamp: time.Now().Add(time.Hour * 24),
+			Root: commitmenttypes.NewMerkleRoot(
+				[]byte("LpGpeyQVLUo9HpdsgJr12NP2eCICspcULiWa5u9udOA="),
+			),
+			NextValidatorsHash: []byte("E30CE736441FB9101FADDAF7E578ABBE6DFDB67207112350A9A904D554E1F5BE"),
```

Comment:
@tbruyelle where are the values for the Merkle root and the next validator coming from?

Reply:
well I think I scavenged them from the `interchain-security` integration tests. The thing is you have to provide well encoded values or else it fails at some point. But the value itself has no meaning for now.